gpytorch/lazy/lazy_tensor.py

```diff
@@ -1678,14 +1678,6 @@ def __mul__(self, other):
         """
         return self.mul(other)
 
-    def __setattr__(self, name, val):
```

Comment:
Can we just delete this method?

Reply:
It's really really slow because of how often it gets called, and the checking it does really only ever comes up when we write a *brand new* LazyTensor (e.g., the error will never actually be raised in production code with existing lazy tensors).

I say we can it until we see 3rd party people (1) contributing their own new LazyTensors and (2) actually run in to this issue.